File: app/src/main/java/com/rafaelmfer/nasaexperience/ui/activity/ActivityMarsWeather.java
package com.rafaelmfer.nasaexperience.ui.activity;

import android.os.Bundle;
import android.view.View;
import android.widget.ImageButton;

import androidx.annotation.Nullable;
import androidx.appcompat.app.AppCompatActivity;
import androidx.recyclerview.widget.LinearLayoutManager;
import androidx.recyclerview.widget.RecyclerView;

import com.rafaelmfer.nasaexperience.R;
import com.rafaelmfer.nasaexperience.model.PostsMars;
import com.rafaelmfer.nasaexperience.ui.adapter.AdapterMars;

import java.util.ArrayList;
import java.util.List;

public class ActivityMarsWeather extends AppCompatActivity {

    RecyclerView recyclerView_mars;
    ImageButton ibMarsWeatherBack;
    List<PostsMars> posts = new ArrayList<>();

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_mars_weather);
        recyclerView_mars = findViewById(R.id.recycler_mars);

        RecyclerView.LayoutManager layoutManager = new LinearLayoutManager(this);
        recyclerView_mars.setLayoutManager(layoutManager);
        newPosts();
        AdapterMars adapterMars = new AdapterMars(posts);
        recyclerView_mars.setAdapter(adapterMars);

        ibMarsWeatherBack = findViewById(R.id.mars_weather_back);
        ibMarsWeatherBack.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                onBackPressed();
            }
        });
    }

    public void newPosts() {
        posts.add(new PostsMars(0, "ObservationsMars", "Pressure",
                "Mars Numbers", "Temperature",
                "Presure"));
        posts.add(new PostsMars(1, "ObservationsMars", "Pressure",
                "Mars Numbers", "Temperature",
                "Presure"));
        posts.add(new PostsMars(2, "ObservationsMars", "Pressure",
                "Mars Numbers", "Temperature",
                "Presure"));
        posts.add(new PostsMars(3, "ObservationsMars", "Pressure",
                "Mars Numbers", "Temperature",
                "Presure"));
        posts.add(new PostsMars(4, "ObservationsMars", "Pressure",
                "Mars Numbers", "Temperature",
                "Presure"));
    }
}

File: app/src/main/java/com/rafaelmfer/nasaexperience/ui/activity/ActivityMarsWeather.kt
package com.rafaelmfer.nasaexperience.ui.activity

import androidx.activity.viewModels
import androidx.lifecycle.Observer
import androidx.recyclerview.widget.LinearLayoutManager
import com.rafaelmfer.nasaexperience.baseviews.ActBind
import com.rafaelmfer.nasaexperience.data.RepositoryDatabase
import com.rafaelmfer.nasaexperience.databinding.ActivityMarsWeatherBinding
import com.rafaelmfer.nasaexperience.debugging.ExceptionHandler
import com.rafaelmfer.nasaexperience.extensions.addMarginTopStatusBarHeight
import com.rafaelmfer.nasaexperience.extensions.setFullScreen
import com.rafaelmfer.nasaexperience.extensions.toJson
import com.rafaelmfer.nasaexperience.model.marsweatherservice.InfoWeather
import com.rafaelmfer.nasaexperience.model.room.InfoWeatherRoom
import com.rafaelmfer.nasaexperience.ui.adapter.AdapterMars
import com.rafaelmfer.nasaexperience.viewmodel.ViewModelMarsWeather

class ActivityMarsWeather : ActBind<ActivityMarsWeatherBinding>(), OnItemClickSunMars {

    override val bindClass by lazy { ActivityMarsWeatherBinding::class.java }
    private val viewModel: ViewModelMarsWeather by viewModels()
    private var sunSet = mutableSetOf<InfoWeather>()

    private val repositoryDatabase by lazy { RepositoryDatabase(this) }

    override fun ActivityMarsWeatherBinding.onBoundView() {
        exceptionHandler = ExceptionHandler::class.java
        setFullScreen()

        marsWeatherBack.apply {
            addMarginTopStatusBarHeight()
            setOnClickListener { onBackPressed() }
        }
        recyclerMars.layoutManager = LinearLayoutManager(this@ActivityMarsWeather)

        viewModel.marsWeatherResponse.observe(this@ActivityMarsWeather, Observer { response ->
            sunSet.run {
                add(response.sunNumber1)
                add(response.sunNumber2)
                add(response.sunNumber3)
                add(response.sunNumber4)
                add(response.sunNumber5)
                add(response.sunNumber6)
                add(response.sunNumber7)
            }
            sunSet.forEachIndexed { index, infoWeather ->
                try {
                    infoWeather.sunName = response.solKeys[index]
                } catch (ex: IndexOutOfBoundsException) {
                    infoWeather.sunName = "1"
                }
            }
            recyclerMars.adapter = AdapterMars(sunSet)
        })
        viewModel.getMarsWeather()
    }

    override fun addPositionSunMar(infoWeather: InfoWeather) {
            val entityRoom = InfoWeatherRoom(
                infoWeather.atmosphericTemperature.toJson(),
                infoWeather.firstUTC,
                infoWeather.horizontalWindSpeed.toJson(),
                infoWeather.atmosphericPressure.toJson(),
                infoWeather.windDirection.toJson(),
                infoWeather.sunName,
                true
            )
        repositoryDatabase.accessMars.insert(entityRoom)
    }

//    override fun removePositionSunMar(infoWeather: InfoWeather) {
//        repositoryDatabase.accessMars.delete(repositoryDatabase.accessMars.findMarsName(infoWeather.sunName))
//    }
}